acceptance/suites/tests/authorization/x509_auth.rb

```diff
@@ -0,0 +1,139 @@
+test_name "(TK-293) TK-AUTH uses certificate extensions for authentication" do
+
+confine :except, :platform => 'windows'
+  
+server = master.puppet['certname']
+ssldir = master.puppet['ssldir']
+confdir = master.puppet['confdir']
+
+teardown do
+  # restore the original tk auth.conf file
+  on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.bak /etc/puppetlabs/puppetserver/conf.d/auth.conf'
+end
+
+step "Backup the tk auth.conf file" do
+  on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.conf /etc/puppetlabs/puppetserver/conf.d/auth.bak'
+end
+
+# Do we have a functioning cert?
+step "Confirm agent can connect with existing cert" do
+  agents.each do |a|
+    if (not_controller(a))
+      rc = on(a,
+              puppet("agent --test --server #{server} --detailed-exitcodes"),
+              {:acceptable_exit_codes => [0,2]})
+    end
+  end
+end
+
+# Not anymore we don't
+step "Revoke and destroy the existing cert on the server" do
+  agents.each do |a| 
+    if (not_controller(a))
+      nodename = fact_on(a, 'fqdn')
+      rc = on(master, 
+              puppet("cert destroy #{nodename}"),
+              {:acceptable_exit_codes => [0,2]})
+    end
+  end
+end
+
+step "HUP the server" do
+  hup_server
+end
+
+# After a server HUP, the agent cert should be rejected
+step "Confirm agent can't connect with existing cert" do
+  agents.each do |a|
+    if (not_controller(a))
+      rc = on(a,
+              puppet("agent --test --server #{server} --detailed-exitcodes"),
+              {:acceptable_exit_codes => [1]})
+    end
+  end
+end
+
+step "Remove the old certs on the agents so they'll make new ones" do
+  agents.each do |a|
+    if (not_controller(a))
+      nodename = fact_on(a, 'fqdn')
```

Comment:
Why not use the beaker host's certname value a.puppet['certname'] rather than performing an additional facter call?